### ISSUE TITLE
Linking error with 'retroarch_git_version'

### ISSUE THREAD

**Issue by tim77** [libretro/RetroArch#12225]
## Description

Can't compile new `1.9.1` version do to Linking error with `retroarch_git_version`.

### Expected behavior

Should compile fine as previous `1.9.0` version.

### Actual behavior

```cpp
/usr/bin/ld: obj-unix/release/retroarch.o: in function 'printf':
/usr/include/bits/stdio2.h:112: undefined reference to 'retroarch_git_version'
/usr/bin/ld: obj-unix/release/retroarch.o: in function 'retroarch_parse_input_and_config':
/builddir/build/BUILD/RetroArch-1.9.1/retroarch.c:34805: undefined reference to 'retroarch_git_version'
/usr/bin/ld: obj-unix/release/retroarch.o: in function 'retroarch_main_init':
/builddir/build/BUILD/RetroArch-1.9.1/retroarch.c:35101: undefined reference to 'retroarch_git_version'
/usr/bin/ld: obj-unix/release/ui/drivers/ui_qt.o: in function 'QString::fromUtf8(char const*, int)':
/usr/include/qt5/QtCore/qstring.h:706: undefined reference to 'retroarch_git_version'
/usr/bin/ld: /usr/include/qt5/QtCore/qstring.h:706: undefined reference to 'retroarch_git_version'
/usr/bin/ld: obj-unix/release/menu/menu_displaylist.o:/builddir/build/BUILD/RetroArch-1.9.1/menu/menu_displaylist.c:1010: more undefined references to 'retroarch_git_version' follow
collect2: error: ld returned 1 exit status
```

[Full build log (Fedora 35)](https://kojipkgs.fedoraproject.org//work/tasks/9380/64929380/build.log).

### Version/Commit

- RetroArch: v1.9.1/cf6ca2a

### Environment information

- OS: Fedora 32-35
- Compiler: gcc 10.2.1, gcc 11.0.1

### Additional info

In Fedora not allowed git cloning for sources so [official tarball](https://github.com/libretro/RetroArch/archive/refs/tags/v1.9.1.tar.gz) was used. Could this be an issue? If so, how this could workarounded and why it works before v1.9.1? Please help.

**Comment by hizzlekizzle** [comment]
Yes, this is known issue and it's also stopping the official snap package from building, as it uses the tarball instead of a git clone, as well.

**Comment by modeler** [comment]
Does anyone know of a work-around for this other than cloning the current git version instead?

**Comment by hizzlekizzle** [comment]
Aside from waiting for another tarball release, it looks like applying this patch resolves it (generated from the commit that fixed it):
https://github.com/libretro/RetroArch/commit/cbec423413b85202131b69c0a314a6c29335e135.patch

**Comment by tim77** [comment]
@hizzlekizzle thanks a lot.

And now this is fixed in `1.9.2`.
